Source/WebCore/PAL/pal/spi/cocoa/WebPrivacySPI.h

```diff
@@ -0,0 +1,104 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(ADVANCED_PRIVACY_PROTECTIONS)
+
+#if __has_include(<WebPrivacy/WebPrivacy.h>)
+#import <WebPrivacy/WebPrivacy.h>
+#else
+
+#import <Foundation/Foundation.h>
+
+typedef NS_ENUM(NSInteger, WPResourceType) {
+    WPResourceTypeTrackerBlockList = 1,
+    WPResourceTypeLinkFilteringData,
+    WPResourceTypeTrackerDomains,
+    WPResourceTypeTrackerNetworkAddresses,
+    WPResourceTypeAllowedLinkFilteringData,
+};
+
+typedef NS_ENUM(NSInteger, WPNetworkAddressVersion) {
+    WPNetworkAddressVersion4 = 4,
+    WPNetworkAddressVersion6 = 6
+};
+
+@interface WPNetworkAddressRange : NSObject
+@property (nonatomic, readonly) WPNetworkAddressVersion version;
+@property (nonatomic, readonly) const struct sockaddr* address;
+@property (nonatomic, readonly) NSUInteger netMaskLength;
+@property (nonatomic, readonly) NSString *owner;
+@property (nonatomic, readonly) NSString *host;
+@end
+
+@interface WPResourceRequestOptions : NSObject
+@property (nonatomic) BOOL afterUpdates;
+@end
+
+@interface WPLinkFilteringRule : NSObject
+@property (nonatomic, readonly) NSString *queryParameter;
+@property (nonatomic, readonly) NSString *domain;
+@end
+
+@interface WPLinkFilteringData : NSObject
+@property (nonatomic, readonly) NSArray<WPLinkFilteringRule *> *rules;
+@end
+
+@interface WPTrackingDomain  : NSObject
+@property (nonatomic, readonly) NSString *host;
+@property (nonatomic, readonly) NSString *owner;
+@property (nonatomic, readonly) BOOL canBlock;
+@end
+
+typedef void (^WPJSONCompletionHandler)(NSData *, NSError *);
+typedef void (^WPNetworkAddressesCompletionHandler)(NSArray<WPNetworkAddressRange *> *, NSError *);
+typedef void (^WPQueryParametersCompletionHandler)(NSArray<NSString *> *, NSError *);
+typedef void (^WPLinkFilteringDataCompletionHandler)(WPLinkFilteringData *, NSError *);
+
+@interface WPResources : NSObject
+
++ (instancetype)sharedInstance;
+
+- (void)requestTrackerBlockerContentExtension:(WPResourceRequestOptions *)options completionHandler:(WPJSONCompletionHandler)completion;
+- (void)requestTrackerQueryParameterBlockerContentExtension:(WPResourceRequestOptions *)options completionHandler:(WPJSONCompletionHandler)completion;
+- (void)requestTrackerQueryParameters:(WPResourceRequestOptions *)options completionHandler:(WPQueryParametersCompletionHandler)completion;
+- (void)requestTrackerNetworkAddresses:(WPResourceRequestOptions *)options completionHandler:(WPNetworkAddressesCompletionHandler)completion;
+- (void)requestLinkFilteringData:(WPResourceRequestOptions *)options completionHandler:(WPLinkFilteringDataCompletionHandler)completion;
+- (void)requestAllowedLinkFilteringData:(WPResourceRequestOptions *)options completionHandler:(void (^)(WPLinkFilteringData *, NSError *))completion;
```

Comment:
Looks like the completion handler here can be `WPLinkFilteringDataCompletionHandler`.